apps/authentication-service/src/resources/clients/clientsRoutesHandlersV1.ts

```diff
@@ -89,3 +90,30 @@ export const clientsDeleteV1 = async (req: Request, res: Response) => {
     }
   }
 };
+
+export const clientsRotateSecrets = async (req: Request, res: Response) => {
+  const clientToken = getClientTokenFromRequest(req);
+
+  try {
+    const client: Client = await rotateClientSecret(clientToken);
+
+    res.json({
+      clientId: client.client_id,
+      clientSecret: client.client_secret,
+    });
+  } catch (error) {
+    if (
+      typeof error === "string" &&
+      error.toUpperCase().includes("NOT FOUND")
+    ) {
+      res.status(404).json({ message: "Client not found" });
+    } else if (
+      error instanceof Error &&
+      error.message.toUpperCase().includes("TOO MANY REQUESTS")
+    ) {
+      res.status(429).json({ message: "Rate limit exceeded" });
+    } else {
+      res.status(500).json({ message: "Unable to rotate client secret" });
+    }
```

Comment:
Instead of duplicating this code I would recommend creating a method which can be used in all places where this is duplicated.

Reply:
Okay, I refactored these checks into a separate method.

Reply:
@Craiting FYI ^